### PR TITLE
new test engine + increase cover

### PR DESCRIPTION
**Pull request by CRoeckhout**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -164,75 +184,76 @@ func ParseFlags(progname string, args []string) (config *CommandLineConfig, outp
...
+func main() {
+	config, output, err := ParseFlags(os.Args[0], os.Args[1:])
+	LogFlags(config, output, err)
+
+	settingsFolderPath, settingsFilePrefix, settingsVariableName := GetConfigFileLocationValue()
+
+	// TODO : aller chercher le nom du fichier de config dans le index.html
+	settingsFilePath, errorDefineFilePath := DefineFilePath(settingsFolderPath, settingsFilePrefix)
+	if errorDefineFilePath != nil {
+		panic(errorDefineFilePath)
```
Comment (fleroy-isagri) [inline main.go:255]: is it best to replace the use of panic by handleError ?

File: main.go [hunk 2]
```diff
@@ -164,75 +184,76 @@ func ParseFlags(progname string, args []string) (config *CommandLineConfig, outp
 
 	var conf CommandLineConfig
 	// -version / --version
-	flags.BoolVar(&conf.version, "version", false, "Display version and exit")
+	flags.BoolVar(&conf.Version, "version", false, "Display version and exit")
 
 	err = flags.Parse(args)
-	if err != nil {
+	isTestCommand := strings.Contains(strings.Join(flags.Args(), ""), "-test") || strings.Contains(strings.Join(flags.Args(), ""), "-ginkgo")
```
Comment (fleroy-isagri) [inline main.go:190]: what is the idea of "-test" or "-ginko" flags ?

**Review by fleroy-isagri — APPROVED** [review]
Great works

File: main.go (hunk 2 above)
Comment (CRoeckhout) [reply main.go:190]: When triggered by the "go test" or "ginkgo" command the args starts with "-test.-timeout=..." or "-ginkgo..."

Comment (fleroy-isagri) [reply main.go:190]: thanks
it will be nice to have this explanation as a comment for this line

**Pull request merged**